godot-core/src/log.rs

```diff
@@ -77,15 +77,31 @@ macro_rules! godot_print {
     };
 }
 
-pub use crate::{godot_error, godot_print, godot_script_error, godot_warn};
+#[macro_export]
+macro_rules! godot_print_rich {
+    ($fmt:literal $(, $args:expr)* $(,)?) => {
+        $crate::log::print(true, &[
+            $crate::builtin::Variant::from(
+                $crate::builtin::GString::from(
+                    format!($fmt $(, $args)*)
+                )
+            )
```

Comment:
The `FromGodot` trait [is implemented for `String`](https://godot-rust.github.io/docs/gdext/master/godot/builtin/meta/trait.FromGodot.html#impl-FromGodot-for-String), no need to go through `GString` here.